### PR TITLE
Backport 2FA edit screen for com_admin

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Backport 2FA edit screen for com_admin from 4.x to 3.x too

### Testing Instructions

- enable atleast one 2fa plugin
- got to your profile in the backend
- notice that you can not setup 2fa there
- apply this patch (or install the packages from here: https://ci.joomla.org/artifacts/joomla/joomla-cms/staging/30751/downloads/35668)
- got to your profile in the backend
- notice that you now can setup 2fa
- please setup 2fa
- logout
- login with 2fa
- remove 2fa
- logout
- login without 2fa

### Actual result BEFORE applying this Pull Request

It is not possible to configure 2fa from com_admin

### Expected result AFTER applying this Pull Request

We can configure 2fa from com_admin

### Documentation Changes Required

None

### CODE REVIEW

**Comment by gogicomputers** [comment]
I have tested this item :white_check_mark: successfully on 2522d7020317c1ab5e790bab2b02ae26bfce0f3f<br /><br />Testet with Joomla 3.9.21. Patch works.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30751">issues.joomla.org/tracker/joomla-cms/30751</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 2522d7020317c1ab5e790bab2b02ae26bfce0f3f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30751">issues.joomla.org/tracker/joomla-cms/30751</a>.</sub>

**Comment by richard67** [comment]
@zero-24 Please fix PHP code style errors reported by drone here [https://ci.joomla.org/joomla/joomla-cms/35668/1/3](https://ci.joomla.org/joomla/joomla-cms/35668/1/3) .

**Comment by richard67** [comment]
PR should not be RTC before PHP code style errors are fixed.

**Comment by zero-24** [comment]
Should be fixed now thanks @richard67

**Comment by richard67** [comment]
@zero-24 Normally you can check drone results yourself. Let me know if you need some information on that.

**Comment by zero-24** [comment]
I have fixed them just a minute ago right?

**Comment by richard67** [comment]
Previous test results were still valid because change after that was only code style of comments => RTC.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30751">issues.joomla.org/tracker/joomla-cms/30751</a>.</sub>

**Comment by richard67** [comment]
@zero-24 I just meant you could check yourself if there are errors after a commit. Just a hint for improvement, nothing bad.

**Comment by richard67** [comment]
New PHPCS error => back to pennding.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30751">issues.joomla.org/tracker/joomla-cms/30751</a>.</sub>

File: administrator/components/com_admin/models/profile.php [hunk 1]
```diff
@@ -159,4 +212,78 @@ public function save($data)
...
+		$model  = new UsersModelUser;
+
+		return $model->setOtpConfig($userId, $otpConfig);
+	}
+
+	/**
+	 * Generates a new set of One Time Emergency Passwords (OTEPs) for a given user.
+	 *
+	 * @param   integer  $userId   The user ID
+	 * @param   integer  $count    How many OTEPs to generate? Default: 10
```
Comment (richard67) [inline administrator/components/com_admin/models/profile.php]:
```suggestion
	 * @param   integer  $userId  The user ID
	 * @param   integer  $count   How many OTEPs to generate? Default: 10
```

**Comment by zero-24** [comment]
Patched thanks :)

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30751">issues.joomla.org/tracker/joomla-cms/30751</a>.</sub>

**Comment by richard67** [comment]
@zero-24 Sorry if I sounded harsh or so. Was not meant like this. I have to learn to be a bit more patient.

**Comment by zero-24** [comment]
All fine :-)

**Pull request merged**